src/lepton/jpgcoder.cc

```diff
@@ -1648,6 +1648,31 @@ class CachingDecoderReader : public DecoderReader
                               size == 0 ? JpegError::errEOF() : JpegError::nil());
     }
 };
+
+
+// Implementation of Sirikata::DecoderReader API, using the provided std::vector
+// as input data stream
+class VectorDecoderReader : public DecoderReader {
+    const std::vector<uint8_t> &input_data;     // buffer holding entire stream contents
+    size_t                      input_pos;      // current reading position in the input_data
+
+public:
+    VectorDecoderReader(const std::vector<uint8_t> &_input_data)
+        : input_data(_input_data), input_pos(0) {
+    }
+    // Read data into provided buffer data[], returning number of bytes read.
+    // It could be less than requested if we reached EOF.
+    std::pair<uint32_t, JpegError> Read(uint8_t *data, unsigned int size) override {
```

Comment:
Let's use one data type for a size.

Reply:
This API defined here: https://github.com/dropbox/lepton/blob/7533355cfc83bcb7d5a3e4947453842d41199c8b/src/io/Reader.hh#L37

Since there are no guarantees that int has 32 bits, I think that we can change this only for all DecoderReader descendants simultaneously

Reply:
Let's do it.